packages/optimizer/lib/transformers/AutoExtensionImporter.js

```diff
@@ -205,31 +200,25 @@ class AutoExtensionImporter {
     }
   }
 
+  /**
+   * @private
+   */
   calculateVersion(extension, extensionName) {
-    const customVersion = this.extensionVersions[extensionName];
     // Let user override default
+    const customVersion = this.extensionVersions[extensionName];
     if (customVersion) {
       this.log_.debug('using custom version for', extensionName, customVersion);
       return customVersion;
     }
-    // Get latest version as specified by validation rules
-    let version = extension.version[extension.version.length - 1];
-    // Get bento component info as these might still be experimental
-    const bentoComponent = this.bentoComponentInfo[extensionName];
-    if (!bentoComponent) {
-      // No bento component
-      return version;
-    }
-    // Bento component is not experimental, nothing to worry about
-    if (!bentoComponent.experimental) {
-      return version;
-    }
-    // Bento component version is not yet known to the validator
-    if (version != bentoComponent.version) {
-      return version;
+    // Get latest non-experimental prod version (as calculated based on experimental flag)
+    const latestProdVersion = this.componentVersions[extensionName];
+    // Also get latest version supported by the validator
+    const latestValidatorVersion = extension.version[extension.version.length - 1];
+    // Verify latest version based on validator support
+    if (latestValidatorVersion < latestProdVersion) {
```

Comment:
This could cause problems later as `0.11` should be a newer than `0.2`, but here it would be false. In other words, I think this should use [`semver`](https://www.npmjs.com/package/semver):

```suggestion
    if (semver.lt(latestValidatorVersion, latestProdVersion)) {
```

Reply:
Good point, thankfully no longer needed with the updated endpoint.